torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -63,6 +66,7 @@ def pre_grad_passes(gm, example_inputs):
     if config.pattern_matcher:
         lazy_init()
         gm = fuse_fx(gm, example_inputs)
+        efficient_conv_bn_eval_graph_transform(gm)
```

Comment:
I'd like to combine this with the `pattern_matcher_pass` stuff below.

This allows us to do all of our pattern matching in a single pass over the graph, rather than having a seperate pass for each optimization we want to add.  This will help compile times and make this pass free for networks that don't have conv.

Reply:
Looks good to me. I can make it part of the `pattern_matcher_pass`.

I have difficulty understanding the `pattern_matcher_pass` implementation, though:
https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/pre_grad.py#L29

```
normalization_pass = PatternMatcherPass(prevent_match_across_mutations=True)
merge_splits_pass = PatternMatcherPass(prevent_match_across_mutations=True)
split_cat_pass = PatternMatcherPass(prevent_match_across_mutations=True)
unbind_stack_pass = PatternMatcherPass(prevent_match_across_mutations=True)

pattern_matcher_passes: List[PatternMatcherPass] = [
    normalization_pass,
    merge_splits_pass,
    split_cat_pass,
    unbind_stack_pass,
]
```

These four passes have the same construction code `PatternMatcherPass(prevent_match_across_mutations=True)`. How are they different and how do they achieve the normalization/merge/split/xxx functionality?

Reply:
Got it. It is in:

https://github.com/pytorch/pytorch/blob/09f7cb0eaf02d69ab4c788b03345198f5c869718/torch/_inductor/fx_passes/split_cat.py#L89

Will do the refactoring.